src/containers/Stories/index.jsx

```diff
@@ -12,10 +14,25 @@ const Stories = () => {
     const fetchPosts = async () => {
       const data = await fetch(_URL);
       const fetchedPosts = await data.json();
-      setPosts(fetchedPosts);
+      //filter posts based on the language
+      const filteredPosts = await fetchedPosts.filter((post) => {
+        return post.tags.toString() === langRef;
+      });
+      setPosts(filteredPosts);
     };
     fetchPosts();
-  }, []);
+  }, [langRef]);
+
+  i18next.on("languageChanged", (lng) => {
+    let ref = i18next.translator.language;
+    if (ref === "en") {
+      setLangRef("10221");
+    } else if (ref === "ar") {
+      setLangRef("38299");
```

Comment:
magic strings please assign all these refs to a variable

Reply:
revised